megamek/src/megamek/client/ui/dialogs/UnitDisplayDialog.java

```diff
@@ -20,36 +20,75 @@
 
 public class UnitDisplayDialog extends JDialog {
     //region Variable Declarations
-    private UnitDisplay unitDisplay;
-
     private final ClientGUI clientGUI;
+    private static final String MSG_TITLE = Messages.getString("ClientGUI.MechDisplay");
+    private static final GUIPreferences GUIP = GUIPreferences.getInstance();
     //endregion Variable Declarations
 
     //region Constructors
-    public UnitDisplayDialog(final JFrame frame, final UnitDisplay unitDisplay,
-                             final ClientGUI clientGUI) {
-        super(frame, Messages.getString("ClientGUI.MechDisplay"), false);
-        setUnitDisplay(unitDisplay);
+    public UnitDisplayDialog(final JFrame frame, final ClientGUI clientGUI) {
+        super(frame, MSG_TITLE, false);
+
+        if (GUIP.getUnitDisplayStartTabbed()) {
+            this.setLocation(GUIP.getUnitDisplayPosX(), GUIP.getUnitDisplayPosY());
+            this.setSize(GUIP.getUnitDisplaySizeWidth(), GUIP.getUnitDisplaySizeHeight());
+        }
+        else {
+            this.setLocation(GUIP.getUnitDisplayNontabbedPosX(), GUIP.getUnitDisplayNontabbedPosY());
+            this.setSize(GUIP.getUnitDisplayNonTabbedSizeWidth(), GUIP.getUnitDisplayNonTabbedSizeHeight());
+        }
+
+        UIUtil.updateWindowBounds(this);
+        this.setResizable(true);
+        this.setFocusable(false);
+        this.setFocusableWindowState(false);
+
+        addWindowListener(new WindowAdapter() {
+            @Override
+            public void windowClosing(WindowEvent evt) {
+                GUIP.setUnitDisplayEnabled(false);
+            }
+        });
+
         this.clientGUI = clientGUI;
     }
     //endregion Constructors
 
-    //region Getters/Setters
-    public UnitDisplay getUnitDisplay() {
-        return unitDisplay;
+    private void savePref() {
+        if ((this.getSize().width * this.getSize().height) > 0) {
+            if (GUIP.getUnitDisplayStartTabbed()) {
+                GUIP.setUnitDisplayPosX(this.getLocation().x);
+                GUIP.setUnitDisplayPosY(this.getLocation().y);
+                GUIP.setUnitDisplaySizeWidth(this.getSize().width);
+                GUIP.setUnitDisplaySizeHeight(this.getSize().height);
+            } else {
+                GUIP.setUnitDisplayNontabbedPosX(this.getLocation().x);
+                GUIP.setUnitDisplayNontabbedPosY(this.getLocation().y);
+                GUIP.setUnitDisplayNonTabbedSizeWidth(this.getSize().width);
+                GUIP.setUnitDisplayNonTabbedSizeHeight(this.getSize().height);
+                clientGUI.getUnitDisplay().saveSplitterLoc();
+            }
+        }
     }
 
-    public void setUnitDisplay(final UnitDisplay unitDisplay) {
-        this.unitDisplay = unitDisplay;
+    @Override
+    protected void processWindowEvent(WindowEvent e) {
+        super.processWindowEvent(e);
+        if ((e.getID() == WindowEvent.WINDOW_DEACTIVATED) || (e.getID() == WindowEvent.WINDOW_CLOSING)) {
+            savePref();
```

Comment:
As in the previous file: Could the savePref() call be moved up to the WindowClosing listener?